src/super_gradients/common/sg_loggers/clearml_sg_logger.py

```diff
@@ -0,0 +1,226 @@
+from super_gradients.common.environment.env_helpers import multi_process_safe
+
+logger = get_logger(__name__)
+
+try:
+    from clearml import Task
+except (ModuleNotFoundError, ImportError, NameError):
+    pass  # no action or logging - this is normal in most cases
+
+
+class ClearMLSGLogger(BaseSGLogger):
+    def __init__(
+        self,
+        project_name: str,
+        experiment_name: str,
+        storage_location: str,
+        resumed: bool,
+        training_params: dict,
+        checkpoints_dir_path: str,
+        tb_files_user_prompt: bool = False,
+        launch_tensorboard: bool = False,
+        tensorboard_port: int = None,
+        save_checkpoints_remote: bool = True,
+        save_tensorboard_remote: bool = True,
+        save_logs_remote: bool = True,
+    ):
+        """
+
+        :param experiment_name:         Used for logging and loading purposes
+        :param s3_path:                 If set to 's3' (i.e. s3://my-bucket) saves the Checkpoints in AWS S3 otherwise saves the Checkpoints Locally
+        :param checkpoint_loaded:       If true, then old tensorboard files will *not* be deleted when tb_files_user_prompt=True
+        :param max_epochs:              Number of epochs planned for this training
+        :param tb_files_user_prompt:    Asks user for Tensorboard deletion prompt.
+        :param launch_tensorboard:      Whether to launch a TensorBoard process.
+        :param tensorboard_port:        Specific port number for the tensorboard to use when launched (when set to None, some free port number will be used)
+        :param save_checkpoints_remote: Saves checkpoints in s3.
+        :param save_tensorboard_remote: Saves tensorboard in s3.
+        :param save_logs_remote:        Saves log files in s3.
+        """
+        self.s3_location_available = storage_location.startswith("s3")
+        super().__init__(
+            project_name,
+            experiment_name,
+            storage_location,
+            resumed,
+            training_params,
+            checkpoints_dir_path,
+            tb_files_user_prompt,
+            launch_tensorboard,
+            tensorboard_port,
+            self.s3_location_available,
+            self.s3_location_available,
+            self.s3_location_available,
+        )
+        self.setup(project_name, experiment_name)
+
+        self.save_checkpoints = save_checkpoints_remote
+        self.save_tensorboard = save_tensorboard_remote
+        self.save_logs = save_logs_remote
+
+    @multi_process_safe
+    def setup(self, project_name, experiment_name):
+        from multiprocessing.process import BaseProcess
+
+        # Prevent clearml modifying os.fork and BaseProcess.run, which can cause a DataLoader to crash (if num_worker > 0)
+        # Issue opened here: https://github.com/allegroai/clearml/issues/790
+        default_fork, default_run = os.fork, BaseProcess.run
+        self.task = Task.init(
```

Comment:
This will cause an ugly crash if clearml is not installed but this function is called.
please check if Task is defined and raise a nice error message (suggest installing clearml).

Reply:
Check added in f8a57a1711901a0a9a18a8dcb8c0fe579af0db4d